snap/squashfs/squashfs.go

```diff
@@ -102,6 +102,32 @@ func (s *Snap) ReadFile(path string) (content []byte, err error) {
 	return ioutil.ReadFile(filepath.Join(unpackDir, path))
 }
 
+// ListDir returns the content of a single directory inside a squashfs snap.
+func (s *Snap) ListDir(path string) ([]string, error) {
+	tmpdir, err := ioutil.TempDir("", "read-dir")
+	if err != nil {
+		return nil, err
+	}
+	defer os.RemoveAll(tmpdir)
+
+	unpackDir := filepath.Join(tmpdir, "unpack")
+	if err := runCommand("unsquashfs", "-i", "-d", unpackDir, s.path, path); err != nil {
```

Comment:
I think path.Match(path.Join(path, "*"), _) might help there then you have just to path.Base the matching lines